src/aspire/basis/polar_2d.py

```diff
@@ -0,0 +1,122 @@
+from aspire.utils.matrix import roll_dim, unroll_dim
+from aspire.utils.matlab_compat import m_reshape
+from aspire.basis import Basis
+
+logger = logging.getLogger(__name__)
+
+
+class PolarBasis2D(Basis):
+    """
+    Define a derived class for polar Fourier representation for 2D images
+    """
+
+    def __init__(self, size, nrad=None, ntheta=None):
+        """
+        Initialize an object for the 2D polar Fourier grid class
+
+        :param size: The shape of the vectors for which to define the grid.
+            Currently only square images are supported.
+        :param nrad: The number of points in the radial dimension.
+        :param ntheta: The number of points in the angular dimension.
+        """
+
+        ndim = len(size)
+        ensure(ndim == 2, 'Only two-dimensional grids are supported.')
+        ensure(len(set(size)) == 1, 'Only square domains are supported.')
+
+        self.nrad = nrad
+        if nrad is None:
+            self.nrad = self.nres // 2
+
+        self.ntheta = ntheta
+        if ntheta is None:
+            # try to use the same number as Fast FB basis
+            self.ntheta = 8 * self.nrad
+
+        super().__init__(size)
+
+    def _build(self):
+        """
+        Build the internal data structure to 2D polar Fourier grid
+        """
+        logger.info('Represent 2D image in a polar Fourier grid')
+
+        self.count = self.nrad * self.ntheta
+        self._sz_prod = self.sz[0] * self.sz[1]
+
+        # precompute the basis functions in 2D grids
+        self.freqs = self._precomp()
+
+    def _precomp(self):
+        """
+        Precomute the polar Fourier grid
+        """
+        omega0 = 2 * np.pi / (2 * self.nrad - 1)
+        dtheta = 2 * np.pi / self.ntheta
+
+        # Note: The size of ntheta can be reduced by half if only evaluate_t function is needed as
+        # in the Matlab implementation, np.zeros((2, self.nrad * self.ntheta//2)).
+        freqs = np.zeros((2, self.nrad * self.ntheta))
+        for i in range(self.ntheta):
+            freqs[0, i * self.nrad: (i + 1) * self.nrad] = np.arange(self.nrad) * np.sin(i * dtheta)
+            freqs[1, i * self.nrad: (i + 1) * self.nrad] = np.arange(self.nrad) * np.cos(i * dtheta)
+
+        freqs *= omega0
+        return freqs
+
+    def evaluate(self, v):
+        """
+        Evaluate coefficients in standard 2D coordinate basis from those in polar Fourier basis
+
+        :param v: A coefficient vector (or an array of coefficient vectors)
+            in polar Fourier basis to be evaluated. The first two dimensions must equal
+            `self.nrad` and `self.ntheta`.
+        :return x: The evaluation of the coefficient vector(s) `x` in standard 2D
+            coordinate basis. This is an array whose first two dimensions equal `self.sz`
+            and the remaining dimensions correspond to dimensions two and higher of `v`.
+        """
+        # TODO: need check the normalization factor and develop unit test
```

Comment:
Why are we missing a normalization factor?

Reply:
The `evaluate_t` corresponding the `cryo_pft`  function in the Matlab code. But `evaluate`  is not defined. I am not sure the normalization factors for both of them are right to satisfy x=evaluate(evaluate_t(x)).

Reply:
I added the normalization factor based on the equations from the website, http://fourier.eng.hmc.edu/e101/lectures/Image_Processing/node6.html
It has no effects on common line and orientation estimation but does have significant effects on satisfying x=evaluate(evaluate_t(x)).

Reply:
The Normalization factor was removed the new commit fefb446.